azurerm/resource_arm_netapp_volume.go

```diff
@@ -0,0 +1,357 @@
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	if props := resp.VolumeProperties; props != nil {
+		d.Set("creation_token", props.CreationToken)
+		d.Set("service_level", props.ServiceLevel)
+		d.Set("subnet_id", props.SubnetID)
+
+		if props.UsageThreshold != nil {
+			d.Set("usage_threshold", *props.UsageThreshold/1073741824)
+		}
+		if props.ExportPolicy != nil {
+			if err := d.Set("export_policy_rule", flattenArmExportPolicyRule(props.ExportPolicy.Rules)); err != nil {
+				return fmt.Errorf("Error setting `export_policy_rule`: %+v", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func resourceArmNetAppVolumeDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Netapp.VolumeClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	accountName := id.Path["netAppAccounts"]
+	poolName := id.Path["capacityPools"]
+	name := id.Path["volumes"]
+
+	_, err = client.Delete(ctx, resourceGroup, accountName, poolName, name)
+	if err != nil {
+		return fmt.Errorf("Error deleting NetApp Volume %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	return waitForNetAppVolumeToBeDeleted(ctx, client, resourceGroup, accountName, poolName, name)
+}
+
+func waitForNetAppVolumeToBeDeleted(ctx context.Context, client *netapp.VolumesClient, resourceGroup, accountName, poolName, name string) error {
+	log.Printf("[DEBUG] Waiting for NetApp Volume Provisioning Service %q (Resource Group %q) to be deleted", name, resourceGroup)
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"200", "202"},
+		Target:  []string{"404"},
+		Refresh: netappVolumeDeleteStateRefreshFunc(ctx, client, resourceGroup, accountName, poolName, name),
+		Timeout: 20 * time.Minute,
+	}
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("Error waiting for NetApp Volume Provisioning Service %q (Resource Group %q) to be deleted: %+v", name, resourceGroup, err)
+	}
+
+	return nil
+}
+
+func netappVolumeDeleteStateRefreshFunc(ctx context.Context, client *netapp.VolumesClient, resourceGroupName string, accountName string, poolName string, name string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		res, err := client.Get(ctx, resourceGroupName, accountName, poolName, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(res.Response) {
+				return nil, "", fmt.Errorf("Error retrieving NetApp Volume %q (Resource Group %q): %s", name, resourceGroupName, err)
+			}
+		}
+
+		if _, err := client.Delete(ctx, resourceGroupName, accountName, poolName, name); err != nil {
+			log.Printf("Error reissuing NetApp Volume %q delete request (Resource Group %q): %+v", name, resourceGroupName, err)
+		}
+
+		return res, strconv.Itoa(res.StatusCode), nil
+	}
+}
+
+func expandArmExportPolicyRule(input []interface{}) *netapp.VolumePropertiesExportPolicy {
```

Comment:
this function name is scoped to the Package not the File - so can we rename this:

```suggestion
func expandArmNetAppVolumeExportPolicyRule(input []interface{}) *netapp.VolumePropertiesExportPolicy {
```

Reply:
Fixed